### DISCUSSION TITLE
Service_role key & public schema.

### DISCUSSION THREAD

**Discussion by HirotakaAseishi** [supabase/supabase#14294] [Questions] [answered]
Hi,
Is it only tables in the public schema that can bypass the RLS with the service_role key?
(I tried with a view with security_invoker=true that pulled the email from the auth schema.)

For example, if we want to handle different users' emails on the server side, we need to put the user's email in some table in the public schema, is it correct?

**Comment by GaryAustin1** [comment, marked as answer]
Service_role key will bypass all RLS.  But it does not have any privileges granted to the auth schema.  It can't access the schema at all so never even gets to RLS on the table.   You can grant it access to private schemas that have RLS on the tables and it will bypass.  

So you can either have a copy of the email in the public schema, or use a security definer rpc function to access the auth email.

**Reply by HirotakaAseishi** [reply]
Thank you so much! My understanding is now clearer.
